File: src/test/kotlin/com/github/mgurov/talks/kotlinfortesting/domain/BusinessLogicTest.java
package com.github.mgurov.talks.kotlinfortesting.domain;

import org.junit.jupiter.api.Test;

import java.math.BigDecimal;
import java.util.Arrays;
import java.util.List;

import static org.assertj.core.api.Assertions.assertThat;

class BusinessLogicTest {

    @Test
    public void shouldSelectByBuyer() {
        //given
        PurchaseOrder expected1 = new PurchaseOrder("product 1", 1, BigDecimal.ONE, "me");
        PurchaseOrder expected2 = new PurchaseOrder("product 2", 2, BigDecimal.ZERO,"me");
        PurchaseOrder notExpected = new PurchaseOrder("product 2", 2, BigDecimal.TEN,"someone else");
        //when
        List<PurchaseOrder> actual = BusinessLogic.selectByBuyer("me", Arrays.asList(expected1, expected2, notExpected));
        //then
        assertThat(actual)
                .containsExactlyInAnyOrder(expected1, expected2);
    }

}

File: src/test/kotlin/com/github/mgurov/talks/kotlinfortesting/domain/BusinessLogicTest.kt
package com.github.mgurov.talks.kotlinfortesting.domain

import org.assertj.core.api.Assertions
import org.junit.jupiter.api.Test
import java.util.*

class BusinessLogicKotlinTest {
    @Test
    fun shouldSelectByBuyer() {
        //given
        val expected1 = PurchaseOrder("product 1", 1, "me")
        val expected2 = PurchaseOrder("product 2", 2, "me")
        val notExpected = PurchaseOrder("product 2", 2, "someone else")
        //when
        val actual = BusinessLogic.selectByBuyer("me", listOf(expected1, expected2, notExpected))
        //then
        Assertions.assertThat(actual)
                .containsExactlyInAnyOrder(expected1, expected2)
    }
}